serde_someip/src/test.rs

```diff
@@ -0,0 +1,202 @@
+use serde::{ser, Serialize};
+
+use crate::{
+    de::from_bytes,
+    error::{Error, Result},
+    ser::to_bytes,
+};
+use serde::de::{self, DeserializeSeed, EnumAccess, MapAccess, SeqAccess, VariantAccess, Visitor};
+use serde::Deserialize;
+#[test]
+fn test_struct() {
+    #[derive(Serialize, Deserialize, PartialEq, Debug)]
+    struct Test {
+        int: u8,
+        bool: bool,
+        float: f32,
+        char: char,
+        str: String,
+        tup: (u8, u8),
+        //seq: Vec<&'static str>,
+        //seq2: Vec<u8>,
+        //seq3: Vec<Vec<u8>>,
+    }
+
+    let test = Test {
+        int: 98,
+        bool: false,
+        float: 342.1274,
+        char: 'c',
+        str: String::from("test"),
+        tup: (1, 3),
+        //seq: vec!["a", "b"],
+        //seq2: vec![12, 32],
+        //seq3: vec![vec![12, 34], vec![23, 43]],
+    };
+
+    let expected = vec![
+        98, 0, 79, 16, 171, 67, 99, 7, 0, 0, 0, 239, 187, 191, 116, 101, 115, 116, 0, 1, 3,
+    ];
+    let len: i32 = expected.len().try_into().unwrap();
+    println!("{}", len);
+    //let j:&[u8]=&expected;
+    let k: &[u8] = expected.as_ref();
+    println!("{:?}", k);
+    // assert_eq!(to_bytes(&test).unwrap(), expected);
+
+    assert_eq!(test, from_bytes(k).unwrap());
+}
+
+#[test]
+fn test_bool() {
```

Comment:
No need to compare against the expected raw bytes here. Those tests are already done in the corresponding serialization tests.  In this file, only perform roundtrip tests.

1. Initialize a type
2. Serialize to bytes
3. Deserialize from the above bytes
5. Compare the original value with the deserialized value.

No need to check the raw bytes in between.  This comment applies to all the tests.